#! /bin/sh /usr/share/dpatch/dpatch-run
## 65-use-proper-environent.dpatch by Sven Mueller <sven@debian.org>
##    most parts originally by Alexander Wirt <formorer@formorer.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: use a sane environment and a proper homedir

@DPATCH@
diff -urNad trunk/spampd /tmp/dpep.y27603/trunk/spampd
--- trunk/spampd	2007-05-22 18:51:53.000000000 +0200
+++ /tmp/dpep.y27603/trunk/spampd	2007-05-22 19:02:20.997755983 +0200
@@ -853,6 +853,7 @@
 	       logsock => \$logsock,
 	       envelopeheaders => \$envelopeheaders,
 	       setenvelopefrom => \$setenvelopefrom,
+	       homedir => \$homedir, 
 	       pfs_uses_format => \$pfs_uses_format,
 	      );
 
@@ -879,6 +880,7 @@
 		   'help|h|?',
 		   'local-only|l',
 		   'config|C=s',
+		   'homedir=s',
 		   'log-rules-hit|rh',
 		   'dose',
 		   'add-sc-header|ash',  # deprecated
@@ -910,6 +912,13 @@
 # my $min_spare_servers = ($children == $maxchildren) ? 0 : 1;
 # my $max_spare_servers = ($min_spare_servers == 0) ? 0 : $maxchildren-1;
 
+#cleanup environment 
+
+$ENV{'PATH'} = '/bin:/usr/bin:/sbin:/usr/sbin';
+delete @ENV{'IFS', 'CDPATH', 'ENV', 'BASH_ENV', 'HOME'};
+
+my $home_dir = $options{'homedir'} || '/var/cache/spampd'; 
+
 my @tmp = split (/:/, $relayhost);
 $relayhost = $tmp[0];
 if ( $tmp[1] ) { $relayport = $tmp[1]; }
@@ -923,6 +932,8 @@
 	$assassin=Mail::SpamAssassin->new({
 		'dont_copy_prefs' => 1,
 		'debug' => $debug,
+		'home_dir_for_helpers' => $home_dir, 
+		'userstate_dir' => $home_dir, 
 		'userprefs_filename' => ${$options{'config'}},
 		'local_tests_only' => $options{'local-only'} || 0 });
 } else {
@@ -930,6 +941,8 @@
 		'dont_copy_prefs' => 1,
 		'debug' => $debug,
 		'local_tests_only' => $options{'local-only'} || 0 });
+		'home_dir_for_helpers' => $home_dir,
+		'userstate_dir' => $home_dir, 
 };
 
 $options{'auto-whitelist'} and eval {
@@ -992,6 +1005,7 @@
 				instance => 0,
 				envelopeheaders => $envelopeheaders,
 				setenvelopefrom => $setenvelopefrom,
+				homedir => \$homedir, 
 	       			pfs_uses_format => $pfs_uses_format,
 			   },
    }, 'SpamPD';
@@ -1474,6 +1488,11 @@
 SpamAssassin's normal configuration files, overriding the latter. Defaults
 to not using a spampd specific configuration file.
 
+=item B<--homedir=directory>
+
+Use the specified directory as home directory for the spamassassin process. 
+Defaults to /var/cache/spampd. 
+
 =item B<--debug> or B<--d>
 
 Turns on SpamAssassin debug messages which print to the system mail log
